Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift

```diff
@@ -350,6 +350,246 @@ extension GenericArgumentSyntax {
   }
 }
 
+extension FunctionCallExprSyntax {
+  @available(
+    *,
+    deprecated,
+    message: "Use the initializer that does not provide default values for leftParen and rightParen."
+  )
+  public init(
```

Comment:
Do we need all variations of `leftParen` and `rightParen` missing? Couldn’t we have one compatibility function that still has the `= nil` default value?

Reply:
I initially intended to implement it as you suggested.
However, when all parameters in an initializer share the same labels and only a default value(`= nil`) is added, the compiler considers them identical and raises a redeclaration error 🤔 
To resolve this, I had to define separate initializers for each case.

Reply:
Oh, that’s unfortunate. But fair enough. Thanks for spelling them out.

Reply:
Thanks for taking the time to review 😊